src/components/MovieForm/Component/MovieForm.tsx

```diff
@@ -1,62 +1,68 @@
-import React from "react";
 import styles from "../Styles/MovieForm.module.css";
 import { Movie } from "../../../types/movie";
 import { genres } from "../../GenreSelect/Component/GenreSelect";
+import { useForm } from "react-hook-form";
 
 interface MovieFormProps {
   initialMovie?: Movie;
   onSubmit: (movie: Movie) => void;
 }
 
 const MovieForm = ({ initialMovie, onSubmit }: MovieFormProps) => {
-  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const formData = new FormData(event.currentTarget);
-    const movie: Movie = {
-      title: formData.get("title") as string,
-      poster_path: formData.get("movieUrl") as string,
-      genres: Array.from(formData.getAll("genre") as Iterable<string>),
-      release_date: formData.get("releaseDate") as string,
-      vote_average: parseFloat(formData.get("rating") as string),
-      runtime: parseInt(formData.get("runtime") as string),
-      overview: formData.get("overview") as string,
-    };
-
-    onSubmit(movie);
-  };
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<Movie>();
 
   return (
     <>
       <div className={styles.container}>
-        <form onSubmit={handleSubmit}>
+        <form onSubmit={handleSubmit(onSubmit)}>
           <div className={styles.row}>
             <div className={styles["col-50"]}>
               <label htmlFor="title">TITLE</label>
               <input
                 id="title"
                 type="text"
-                name="title"
                 defaultValue={initialMovie?.title}
                 placeholder="Movie title"
+                {...register("title", { required: "This field is required" })}
+                className={errors.title ? styles.errorInput : undefined}
               />
-
+              {errors.title && (
+                <p className={styles.error}>{errors.title.message}</p>
+              )}
               <label htmlFor="movieUrl">MOVIE URL</label>
               <input
                 id="movieUrl"
                 type="text"
-                name="movieUrl"
                 defaultValue={initialMovie?.poster_path}
                 placeholder="https://"
+                {...register("poster_path", {
+                  required: "This field is required",
+                  pattern: {
+                    value:
+                      /https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\+.~#?&\/=]*)/,
```

Comment:
regexp from my point of view can be extracted from component. It makes JSX harder to read